src/main/java/org/jabref/logic/citationstyle/CSLAdapter.java

```diff
@@ -108,8 +117,46 @@ private static CSLItemData bibEntryToCSLItemData(BibEntry bibEntry, BibDatabaseC
 
             Optional<BibEntryType> entryType = entryTypesManager.enrich(bibEntry.getType(), bibDatabaseContext.getMode());
 
-            Set<Field> fields = entryType.map(BibEntryType::getAllFields).orElse(bibEntry.getFields());
+            if (bibEntry.getType().equals(StandardEntryType.Article)) {
+                // Patch bibEntry to contain the right BibTeX (not BibLaTeX) fields
+                // Note that we do not need to convert from "pages" to "page", because CiteProc already handles it
+                // See BibTeXConverter
+                if (bibDatabaseContext.isBiblatexMode()) {
+                    // Move "number" to "issue" if "issue" does not exist
+                    Optional<String> numberField = bibEntry.getField(StandardField.NUMBER);
+                    numberField.ifPresent(number -> {
+                        if (!bibEntry.hasField(StandardField.ISSUE)) {
+                            bibEntry.setField(StandardField.ISSUE, number);
+                            bibEntry.clearField(StandardField.NUMBER);
+                        }
+                    });
+                    bibEntry.getField(StandardField.EID).ifPresent(eid -> {
+                        bibEntry.setField(StandardField.NUMBER, eid);
+                        bibEntry.clearField(StandardField.EID);
+                    });
+                } else {
+                    // BibTeX mode
+                    bibEntry.getField(StandardField.NUMBER).ifPresent(number -> {
+                        bibEntry.setField(StandardField.ISSUE, number);
+                        bibEntry.clearField(StandardField.NUMBER);
+                    });
+                    bibEntry.getField(StandardField.PAGES).ifPresent(pages -> {
+                        if (pages.toLowerCase(Locale.ROOT).startsWith("article ")) {
+                            pages = pages.substring("Article ".length());
+                            bibEntry.setField(StandardField.NUMBER, pages);
+                        }
+                    });
+                    bibEntry.getField(StandardField.EID).ifPresent(eid -> {
+                        if (!bibEntry.hasField(StandardField.PAGES)) {
```

Comment:
`hasField` here, might lead to tests failing because if `eid` is present, but `pages` field is NOT present, eid will not get rendered, right?

Reply:
The more I think about it: Some tests might fail, simply, because we make the preview render `eid` for bibtex mode. Since eid is not part of the bibtex standard, I came up with tests that basically ignore eid, in bibtex mode and only take eid into account if JabRef users use BibLaTeX mode. So the tests would need to be changed if we include lines 149 - 154. We could remove these lines altogether and see how tests play out without.